ssl/statem/statem_clnt.c

```diff
@@ -1130,12 +1133,24 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
             }
         }
 
+        s->session->ssl_version = s->version;
         s->session->session_id_length = session_id_len;
         /* session_id_len could be 0 */
         memcpy(s->session->session_id, PACKET_data(&session_id),
                session_id_len);
     }
 
+    /* Session version and negotiated protocol version should match
+     * (This may not be the case when reusing old session)
+     */
+    if (s->version != s->session->ssl_version) {
+        al = SSL_AD_PROTOCOL_VERSION;
+
+        /* XXX(indutny): introduce new error code */
```

Comment:
Probably someone forgot to run `make errors` previously. Just commit the changes relevant to you.